neqo-transport/src/tracking.rs

```diff
@@ -453,42 +486,38 @@ impl RecvdPackets {
         };
 
         if ack_eliciting {
-            self.pkts_since_last_ack += 1;
-
-            // Send ACK right away if out-of-order
-            // On the first in-order ack-eliciting packet since sending an ACK,
-            // set a delay.
-            // Count packets until we exceed MAX_UNACKED_PKTS, then remove the
-            // delay.
-            if pn != next_in_order_pn {
-                self.ack_time = Some(now);
-            } else if self.space == PacketNumberSpace::ApplicationData {
-                match &mut self.pkts_since_last_ack {
-                    0 => unreachable!(),
-                    1 => self.ack_time = Some(now + ACK_DELAY),
-                    x if *x > MAX_UNACKED_PKTS => self.ack_time = Some(now),
-                    _ => debug_assert!(self.ack_time.is_some()),
-                }
+            self.unacknowledged_count += 1;
+
+            let immediate_ack = self.space != PacketNumberSpace::ApplicationData
+                || (pn != next_in_order_pn && !self.ignore_order)
+                || self.unacknowledged_count > self.unacknowledged_tolerance;
+
+            let ack_time = if immediate_ack {
+                now
             } else {
-                self.ack_time = Some(now);
-            }
-            qdebug!([self], "Set ACK timer to {:?}", self.ack_time);
+                self.ack_time.unwrap_or_else(|| now + self.ack_delay)
```

Comment:
Do you think that I should be using `min()` here?  I think that this is OK without that.  If the delay decreases, it will just not take effect for a tiny amount of time.  Do you think a comment might help?

Reply:
This comment looks fine. Thanks.